From 301d565b0692a27ab5f4b269b6b3fe9f5d2e59d8 Mon Sep 17 00:00:00 2001
From: Chris PeBenito <chpebeni@linux.microsoft.com>
Date: Mon, 14 Apr 2025 16:49:32 -0400
Subject: [PATCH 35/35] Revert "users: Move unconfined_u definition to
 unconfined module."

This reverts commit ca3698d543c22dbc78c4c491133405754a9f8a3f.

Azure Linux 3 does not have a toolchain with the compiler fix.
See https://github.com/SELinuxProject/refpolicy/issues/829
---
 policy/modules/kernel/kernel.te     |  3 +++
 policy/modules/system/unconfined.te | 14 --------------
 policy/users                        |  7 +++++++
 3 files changed, 10 insertions(+), 14 deletions(-)

diff --git a/policy/modules/kernel/kernel.te b/policy/modules/kernel/kernel.te
index a3c854bef..22c68745a 100644
--- a/policy/modules/kernel/kernel.te
+++ b/policy/modules/kernel/kernel.te
@@ -33,6 +33,9 @@ role sysadm_r;
 role staff_r;
 role user_r;
 
+# here until order dependence is fixed:
+role unconfined_r;
+
 ifdef(`enable_mls',`
 	role secadm_r;
 	role auditadm_r;
diff --git a/policy/modules/system/unconfined.te b/policy/modules/system/unconfined.te
index be0b51d82..760c87207 100644
--- a/policy/modules/system/unconfined.te
+++ b/policy/modules/system/unconfined.te
@@ -8,9 +8,6 @@ policy_module(unconfined)
 # usage in this module of types created by these
 # calls is not correct, however we dont currently
 # have another method to add access to these types
-
-role unconfined_r;
-
 userdom_base_user_template(unconfined)
 userdom_manage_home_role(unconfined_r, unconfined_t)
 userdom_manage_tmp_role(unconfined_r, unconfined_t)
@@ -257,14 +254,3 @@ unconfined_domain_noaudit(unconfined_execmem_t)
 optional_policy(`
 	unconfined_dbus_chat(unconfined_execmem_t)
 ')
-
-########################################
-#
-# Unconfined seuser
-#
-
-ifdef(`direct_sysadm_daemon',`
-        gen_user(unconfined_u, unconfined, unconfined_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
-',`
-        gen_user(unconfined_u, unconfined, unconfined_r, s0, s0 - mls_systemhigh, mcs_allcats)
-')
diff --git a/policy/users b/policy/users
index 25402afd8..ca203758c 100644
--- a/policy/users
+++ b/policy/users
@@ -28,6 +28,13 @@ gen_user(user_u, user, user_r, s0, s0)
 gen_user(staff_u, staff, staff_r sysadm_r ifdef(`enable_mls',`secadm_r auditadm_r'), s0, s0 - mls_systemhigh, mcs_allcats)
 gen_user(sysadm_u, sysadm, sysadm_r, s0, s0 - mls_systemhigh, mcs_allcats)
 
+# Until order dependence is fixed for users:
+ifdef(`direct_sysadm_daemon',`
+        gen_user(unconfined_u, unconfined, unconfined_r system_r, s0, s0 - mls_systemhigh, mcs_allcats)
+',`
+        gen_user(unconfined_u, unconfined, unconfined_r, s0, s0 - mls_systemhigh, mcs_allcats)
+')
+
 #
 # The following users correspond to Unix identities.
 # These identities are typically assigned as the user attribute
-- 
2.49.0

